enterprise/coderd/prebuilds/metricscollector_test.go

```diff
@@ -164,31 +164,6 @@ func TestMetricsCollector(t *testing.T) {
 			templateDeleted: []bool{false},
 			eligible:        []bool{false},
 		},
-		{
-			name:         "deleted templates never desire prebuilds",
-			transitions:  allTransitions,
-			jobStatuses:  allJobStatuses,
-			initiatorIDs: []uuid.UUID{agplprebuilds.SystemUserID},
-			ownerIDs:     []uuid.UUID{agplprebuilds.SystemUserID, uuid.New()},
-			metrics: []metricCheck{
-				{prebuilds.MetricDesiredGauge, ptr.To(0.0), false},
-			},
-			templateDeleted: []bool{true},
-			eligible:        []bool{false},
-		},
-		{
-			name:         "running prebuilds for deleted templates are still counted, so that they can be deleted",
```

Comment:
I understand that this test is for metrics. But removing this does make me wonder if the named business rule "running prebuilds for deleted templates are still counted, so that they can be deleted" still applies in the reconciliation loop. We do need to delete running prebuilds for deleted templates, don't we?

Or is it rather the case that you can't delete a template until its prebuilds are deleted?